### ISSUE TITLE
Improve Git Commit Message

### ISSUE THREAD

**Issue by fargozhu** [SUSE/kubecf#125]
**Is your feature request related to a problem? Please describe.**

the current git commit messages cannot be used to produce automated release notes so it forces an unpleasant human activity - digging and sometimes guesswork.

**Describe the solution you'd like**

Enforce a list of patterns (type and github issue associated,...), language (use imperative form) and format (_summary_ (50 chars) and _body_ (72 chars) separated by one blank line, number of chars, etc).
```
[BUG][#123] - Fix git commit summary

Here it goes a more elaborated explanation if needed and that helps
the developers reviewing and/or to understand the reason behind this
commit message
```

**Describe alternatives you've considered**

Create scripts that would enforce the git commit message policies and use them within githooks and/or within CI platform.

**Additional context**

Detailed information will be captured on a ADRS.

**Comment by fargozhu** [comment]
tl;dr https://chris.beams.io/posts/git-commit/

**Comment by loewenstein** [comment]
@fargozhu May I suggest ADRS to give this and similar decisions a persistent storage?

See for example https://github.com/cloudfoundry/cf-k8s-networking/blob/master/doc/adrs/0001-record-architecture-decisions.md and the sibling files.

**Comment by fargozhu** [comment]
> @fargozhu May I suggest ADRS to give this and similar decisions a persistent storage?
> 
> See for example https://github.com/cloudfoundry/cf-k8s-networking/blob/master/doc/adrs/0001-record-architecture-decisions.md and the sibling files.

**Comment by fargozhu** [comment]
@loewenstein sounds a good lightly way to document 👍 . Let's use this one as starting point...

**Comment by loewenstein** [comment]
@fargozhu well, I'd use a copy of the linked one as the very first. It nicely documents the method, decision to use it and has a link to an optional cli tool that exists...

But then yes, I'd do the commit msg decision immediately.
